Mathlib/LinearAlgebra/Dimension/StrongRankCondition.lean

```diff
@@ -542,3 +547,19 @@ theorem mem_span_set_iff_exists_finsupp_le_finrank :
     exact mem_span_set.mpr ⟨c, hcs, hx⟩
 
 end Submodule
+
+section IsQuadraticAlgebra
+
+/--
+An extension of rings `R ⊆ S` is quadratic if `S` is a free `R`-algebra of rank `2`.
+-/
+-- TODO. use this in connection with `NumberTheory.Zsqrtd`
+class IsQuadraticAlgebra (R S : Type*) [CommSemiring R] [StrongRankCondition R] [Semiring S]
+    [Algebra R S] extends Module.Free R S where
+  finrank_eq_two' : Module.finrank R S = 2
```

Comment:
Maybe I missed prior discussion? But can't you just assume `Module` instead of `Algebra`, and then everything still works?
```suggestion
class IsQuadraticAlgebra (R M : Type*) [CommSemiring R] [StrongRankCondition R] [AddMonoid M]
    [Module R M] extends Module.Free R M where
  finrank_eq_two' : Module.finrank R M = 2
```

Reply:
Indeed. I made the changes. 

Thanks!